compiler/semantic/expr.go

```diff
@@ -224,6 +224,30 @@ func semExpr(scope *Scope, e ast.Expr) (dag.Expr, error) {
 			Kind:    "MapExpr",
 			Entries: entries,
 		}, nil
+	case *ast.OverExpr:
+		exprs, err := semExprs(scope, e.Exprs)
+		if err != nil {
+			return nil, err
+		}
+		if e.Scope == nil {
+			return nil, errors.New("over expression missing scope in AST")
+		}
+		scope.Enter()
+		defer scope.Exit()
+		locals, err := semVars(scope, e.Locals)
+		if err != nil {
+			return nil, err
+		}
+		seq, err := semSequential(nil, scope, e.Scope, nil, nil)
```

Comment:
The package context documentation discourages passing a nil context.  (This requires changing `if ctx == nil {` to `if adapter == nil {` in `semProc`.)
```suggestion
		seq, err := semSequential(ctx, scope, e.Scope, nil, nil)
```

Reply:
Context is available here as it is not needed along this code path so I get the convention but I think it would be best to pass nil.  If the nil gets ends up getting used below somehow thats a bug that I would want to panic on.  But I will change the check from ctx to adaptor != nil.